tests/test_auto_parallel/test_resnet_block_runtime.py

```diff
@@ -0,0 +1,170 @@
+            identity = self.downsample(x)
+
+        out = self.relu(out)
+
+        return out
+
+
+def check_apply_bottleneck(rank, world_size, port):
+    disable_existing_loggers()
+    launch(config={}, rank=rank, world_size=world_size, host='localhost', port=port, backend='nccl')
+    input = torch.rand(256, 64, 64, 64).cuda()
+    physical_mesh_id = torch.arange(0, 4)
+    mesh_shape = (2, 2)
+    # [[0, 1]
+    #  [2, 3]]
+    device_mesh = DeviceMesh(physical_mesh_id, mesh_shape, init_process_group=False)
+    entire_shape = torch.Size((4, 4, 8, 8))
+
+    tracer = ColoTracer()
+    model = Bottleneck(64, 64, 1, norm_layer=torch.nn.modules.batchnorm.BatchNorm2d).cuda()
+    # graph():
+    #     %x : torch.Tensor [#users=1] = placeholder[target=x]
+    #     %conv1 : [#users=1] = call_module[target=conv1](args = (%x,), kwargs = {})
+    #     %bn1 : [#users=1] = call_module[target=bn1](args = (%conv1,), kwargs = {})
+    #     %relu : [#users=1] = call_module[target=relu](args = (%bn1,), kwargs = {})
+    #     %conv2 : [#users=1] = call_module[target=conv2](args = (%relu,), kwargs = {})
+    #     %bn2 : [#users=1] = call_module[target=bn2](args = (%conv2,), kwargs = {})
+    #     %relu_1 : [#users=1] = call_module[target=relu](args = (%bn2,), kwargs = {})
+    #     %conv3 : [#users=1] = call_module[target=conv3](args = (%relu_1,), kwargs = {})
+    #     %bn3 : [#users=1] = call_module[target=bn3](args = (%conv3,), kwargs = {})
+    #     %relu_2 : [#users=1] = call_module[target=relu](args = (%bn3,), kwargs = {})
+    #     return relu_2
+    input_sample = {'x': torch.rand(256, 64, 224, 224).to('meta')}
+    cuda_rng_state = torch.cuda.get_rng_state()
+    origin_output = model(input)
+    graph = tracer.trace(root=model, meta_args=input_sample)
+    gm = GraphModule(model, graph, model.__class__.__name__)
+    gm.recompile()
+    solver_options = SolverOptions(fast=True)
+    strategies_constructor = StrategiesConstructor(graph, device_mesh, solver_options)
+    strategies_constructor.build_strategies_and_cost()
+
+    cost_graph = CostGraph(strategies_constructor.leaf_strategies)
+    cost_graph.simplify_graph()
+    graph_analyser = GraphAnalyser(gm)
+    solver = Solver(gm.graph, strategies_constructor, cost_graph, graph_analyser)
+    ret = solver.call_solver_serialized_args()
+    solution = list(ret[0])
+    print(solution)
+    device_mesh.process_groups_dict = device_mesh.create_process_groups_for_logical_mesh()
+    sharding_spec_dict, origin_spec_dict = solution_annotatation_pass(gm, solution, device_mesh)
+    shape_consistency_pass(gm)
+    gm.recompile()
+    nodes = [node for node in gm.graph.nodes]
+    # TODO: wrap the gm to avoid the influence of the user training code
+    torch.cuda.set_rng_state(cuda_rng_state)
+    output = gm(input, sharding_spec_dict, origin_spec_dict)
+    assert output.shape == origin_output.shape
+    assert output.equal(origin_output)
+
+
+@pytest.mark.dist
```

Comment:
Mark this with `run_on_env_flag` if it takes a while.

Reply:
Done

Reply:
I don't see it here?

Reply:
Sorry, forget to push it.